internal/pipe/webhook/webhook.go

```diff
@@ -0,0 +1,158 @@
+	var cfg Config
+	if err := env.Parse(&cfg); err != nil {
+		return fmt.Errorf("announce: failed to announce to webhook: %w", err)
+	}
+
+	endpointURLConfig, err := tmpl.New(ctx).Apply(ctx.Config.Announce.Webhook.EndpointURL)
+	if err != nil {
+		return fmt.Errorf("announce: failed to announce to webhook: %w", err)
+	}
+	if len(endpointURLConfig) == 0 {
+		return errors.New("announce: failed to announce to webhook: no endpoint url")
+	}
+
+	reg := regexp.MustCompile(EndpointRegex)
+	if !reg.MatchString(endpointURLConfig) {
+		return errors.New("announce: failed to announce to webhook: malformed endpoint url")
+	}
+	if _, err := url.ParseRequestURI(endpointURLConfig); err != nil {
+		return fmt.Errorf("announce: failed to announce to webhook: %w", err)
+	}
+	endpointURL, err := url.Parse(endpointURLConfig)
+	if err != nil {
+		return fmt.Errorf("announce: failed to announce to webhook: %w", err)
+	}
+
+	msg, err := tmpl.New(ctx).Apply(ctx.Config.Announce.Webhook.MessageTemplate)
+	if err != nil {
+		return fmt.Errorf("announce: failed to announce to webhook: %s", err)
+	}
+
+	log.Infof("posting: '%s'", msg)
+	customTransport := http.DefaultTransport.(*http.Transport).Clone()
+
+	if !ctx.Config.Announce.Webhook.CheckCert {
+		customTransport.TLSClientConfig = &tls.Config{
+			InsecureSkipVerify: true,
+		}
+	}
+
+	client := &http.Client{
+		Transport: customTransport,
+	}
+
+	req, err := http.NewRequest(http.MethodPost, endpointURL.String(), strings.NewReader(msg))
+	if err != nil {
+		return fmt.Errorf("announce: failed to announce to webhook: %w", err)
+	}
+	req.Header.Add(ContentTypeHeaderKey, ctx.Config.Announce.Webhook.ContentType)
+	req.Header.Add(UserAgentHeaderKey, UserAgentHeaderValue)
+
+	if cfg.BasicAuthHeader != "" {
+		log.Debugf("set basic auth header")
+		req.Header.Add(AuthorizationHeaderKey, cfg.BasicAuthHeader)
+	} else if cfg.BearerTokenHeader != "" {
+		log.Debugf("set bearer token header")
+		req.Header.Add(AuthorizationHeaderKey, cfg.BearerTokenHeader)
+	}
+
+	for key, value := range ctx.Config.Announce.Webhook.Headers {
+		log.Debugf("Header Key %s / Value %s", key, value)
+		req.Header.Add(key, value)
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		return fmt.Errorf("announce: failed to announce to webhook: %w", err)
+	}
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case http.StatusOK, http.StatusCreated, http.StatusAccepted, http.StatusNoContent:
+		log.Infof("Post OK: '%v'", resp.StatusCode)
+		body, _ := io.ReadAll(resp.Body)
+		log.Infof("Response : %v\n", string(body))
+		return nil
+	case http.StatusBadRequest:
+		return ErrHeaderMissing
+	case http.StatusUnauthorized:
+		return ErrClientAuthenticationError
+	case http.StatusForbidden:
+		return ErrForbidden
+	case http.StatusNotFound:
+		return ErrNotFound
+	case http.StatusUnprocessableEntity:
+		return ErrUnprocessableEntityError
+	case http.StatusTooManyRequests:
+		return ErrTooManyRequest
+	default:
+		return errors.New(resp.Status)
```

Comment:
I would just wrap them all with something like:

```
fmt.Errorf("request failed with status %v", resp.Status)
```

and drop all these custom errors... we are not handling them anyway

Reply:
done! Thanks for the hint, you are right!